src/crimson/osd/pg_recovery.cc

```diff
@@ -258,10 +258,10 @@ std::optional<crimson::osd::blocking_future<>> PGRecovery::recover_missing(
   const hobject_t &soid, eversion_t need)
 {
   if (pg->get_peering_state().get_missing_loc().is_deleted(soid)) {
-    return pg->get_recovery_backend()->get_recovering(soid).make_blocking_future(
+    return pg->get_recovery_backend()->add_recovering(soid).make_blocking_future(
```

Comment:
Maybe this line can be moved out of the "if (pg->get_peering_state().get_missing_loc().is_deleted(soid))" block? :-)